social_protection/workflows/utils.py

```diff
@@ -53,7 +54,8 @@ def validate_dataframe_headers(self, is_update=False):
         4. If action is data upload then 'ID' unique identifier is required as well.
         """
         df_headers = set(self.df.columns)
-        schema_properties = set(self.schema.get('properties', {}).keys())
+        schema = json.loads(self.schema)
```

Comment:
@weilu this means that the schema is not loaded before, maybe we should do something like 
```suggestion
        schema = json.loads(self.schema) if isinstance(self.schema, str) else self.schema
```